spi-ch341.c

```diff
@@ -497,22 +497,24 @@ static int match_gpiochip_parent(struct gpio_chip *gc, void *data)
 static int ch341_spi_probe(struct platform_device *pdev)
 {
 	struct ch341_ddata *ch341 = dev_get_drvdata(pdev->dev.parent->parent);
-	struct spi_master *master;
+	struct spi_controller *master;
 	struct ch341_spi *dev;
 	int ret;
 
 	master = spi_alloc_master(&pdev->dev, sizeof(*dev));
 	if (!master)
 		return -ENOMEM;
 
-	dev = spi_master_get_devdata(master);
+	dev = spi_controller_get_devdata(master);
 
 	dev->master = master;
 	dev->ch341 = ch341;
 	platform_set_drvdata(pdev, dev);
 
 	/* Find the parent's gpiochip */
-	dev->gpiochip = gpiochip_find(pdev->dev.parent, match_gpiochip_parent);
+	dev->gpiochip = gpio_device_get_chip(
+		gpio_device_find(pdev->dev.parent, match_gpiochip_parent)
```

Comment:
I had a similar pacth but never tested it. However gpio_device_find may return an error, which gpio_device_get_chip() may not like. 
Also I don't see the required gpio_device_put().

Reply:
See https://github.com/frank-zago/ch341-i2c-spi-gpio/commit/a8ae364b126ea15556822d61ad061f94000fc73a (untested)

Reply:
Yeah https://github.com/frank-zago/ch341-i2c-spi-gpio/commit/a8ae364b126ea15556822d61ad061f94000fc73a looks entirely better

Reply:
Can you integrate it into your PR? I just don't have time to test it.